### ISSUE TITLE
If $id contains spaces, getSchema cannot get properties

### ISSUE THREAD

**Issue by aliyss** [ajv-validator/ajv#2286]
**What version of Ajv are you using? Does the issue happen if you use the latest version?**
Version: 8.12.0 (Current version)

**Ajv options object**

```javascript
new Ajv({ allErrors: true, strict: false, coerceTypes: true})
```

**JSON Schema**

```javascript
// Working
{
  "$id": "ItemSchema",
  "properties": {
      "test": {
         // Schema
      }
   }
}
// Not Working
{
  "$id": "Item Schema",
  "properties": {
      "test": {
         // Schema
      }
   }
}
```

**Sample data**

```javascript
// Data is not relevant, works correctly if $id doesn't contain spaces
```

**Your code**

```javascript
let scope = "Item Schema#/properties/test"
const ajv = new Ajv({ allErrors: true, strict: false, coerceTypes: true })
ajv.addSchema(schema)
const validate = ajv.getSchema(scope)

if (!validate) {
   throw Error()
}
```
Runkit: https://runkit.com/aliyss/ajv-issue

**Validation result, data AFTER validation, error messages**

```
Error
```

**What results did you expect?**

Same result as when using ItemSchema as the $id

**Are you going to resolve the issue?**
I can currently work around it by just renaming, but thought it might be of interest.
It is also completely possible that this is intended and that $id shouldn't even contain spaces.

**Comment by epoberezkin** [comment]
$id should not allow spaces I believe, it’s certainly not a bug.

**Comment by epoberezkin** [comment]
It’s either by the spec - TBC, or, possibly, an intended limitation.

**Comment by aliyss** [comment]
Thanks for the clarification.

About the TBC, I found that $id follows: https://json-schema.org/understanding-json-schema/structuring.html#retrieval-uri

And assuming URIs don't allow unencoded spaces this makes sense.

If I may nitpick a bit: I noticed, that in ajv it will allow you to ``(1) getSchema('Item Schema')``, but not ``(2) getSchema('Item Schema#/properties/test')``.

I did waste some time getting around to finding out what the issue was since (1) worked out fine, but (2) didn't. Would it maybe make sense to have it consistent across or is this intended behaviour?

If you wish I could make a PR. Just need a pointer to which file(s) I can consider editing.
